### PR TITLE
HLS backend update

### PR DESCRIPTION
**Pull request by sjalander**
NB! This PR should not be merged until the jlm-eval-suite has been updated such that a proper commit can be used for the HLS CI.

The HLS backend has been worked on independently of the master branch. This commit is to bring the updated HLS backend up to date with master.

To achieve this, the following three commits that are currently part of the master branch has not been applied to the HLS backend. These commits should be updated and applied at a later time.

beff174d657d4df27a51a5e4371ef8171664f4fb
Clean up HLS unused state removal transformation (#293)

92c65e65c9b533a708cdbd8fe06db6462e332124
Clean up HLS dead node elimination transformation (#294)

33580bba4475ba1f93462a0414fa5a1da9a78395
Clean up HLS gamma conversion (#291)

### CODE REVIEW

**Comment by phate** [comment]
@sjalander It is not even compiling.

**Comment by sjalander** [comment]
@phate I don't know why the CI performs more stringent source code checks, e.g., unused variables, than my local setup on Idun.
There are unfortunately new memory leaks and I currently don't have more time to spend on this.

**Comment by phate** [comment]
@sjalander I can take a look eventually.

**Comment by phate** [comment]
@sjalander It is not just valgrind that is complaining, but also the HLS test suite acts up.

**Comment by sjalander** [comment]
@phate That's a known issue by the builders on github that seems to run out of resources for histogram. I'll take that test out in jlm-eval-suite

File: jlm/llvm/opt/alias-analyses/Optimization.hpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #define JLM_LLVM_OPT_ALIAS_ANALYSES_OPTIMIZATION_HPP
 
 #include <jlm/llvm/opt/alias-analyses/AliasAnalysis.hpp>
+#include <jlm/llvm/opt/alias-analyses/MemoryNodeProvider.hpp>
```
Comment (phate) [inline jlm/llvm/opt/alias-analyses/Optimization.hpp:11]: @sjalander Is this include really necessary. I have my doubts about it.

Comment (phate) [reply jlm/llvm/opt/alias-analyses/Optimization.hpp:11]: Nevermind, looking closer at it, I think it is necessary.

**Comment by phate** [comment]
@sjalander Since the valgrind issue is resolved now, could you create a PR with the other left-over new files?

**Review by phate — CHANGES_REQUESTED** [review]
@sjalander: Can we clarify what it is about `mem-conv.cpp`? There is a cpp file, but no header. Does this mean that this pass is not used and can be removed? Is the header only missing?

**Comment by sjalander** [comment]
@phate mem_conv.hpp was included by mistake in the first PR that added files, i.e., it's already in master.

File: jlm/hls/Makefile.sub [hunk 2]
```diff
@@ -54,6 +56,8 @@ libhls_HEADERS = \
 	jlm/hls/backend/rvsdg2rhls/DeadNodeElimination.hpp \
 	jlm/hls/backend/rvsdg2rhls/distribute-constants.hpp \
 	jlm/hls/backend/rvsdg2rhls/GammaConversion.hpp \
+	jlm/hls/backend/rvsdg2rhls/instrument-ref.hpp \
+	jlm/hls/backend/rvsdg2rhls/mem-conv.hpp \
```
Comment (sjalander) [inline jlm/hls/Makefile.sub:60]: mem-conv.hpp is already in master

**Review by phate — APPROVED** [review]

**Pull request merged**